Fix godoc display issues, other doc fixes

c633 opened the pull request.

masomel commented:
@arlolra @Liamsi Should we add documentation to the executables' `main` package? Original discussion: https://github.com/coniks-sys/coniks-go/pull/131#discussion_r89150645

c633 commented:
@masomel : I rebased this branch with `master` and moved some subpackages to `/internal` in bdcf27b6fe1674eee244bd0c75b8b103aaba3cb4. What do you think? godoc will not show `bar` package if it is an internal package of `foo` (`foo/internal/bar`).

You can use `git reset --soft origin/fix-doc` in your local branch to avoid conflict.

Liamsi commented:
> @arlolra @Liamsi Should we add documentation to the executables' main package? Original discussion: #131 (comment)

I agree with @c633 that this isn't really necessary.

c633 commented:
Any further comments on this?

masomel commented:
> Any further comments on this?

Let me do one more pass over this.

arlolra commented on merkletree/proof.go:
```diff
@@ -114,6 +114,11 @@ func (ap *AuthenticationPath) Verify(key, value, treeHash []byte) bool {
+// between the leaf index and the lookup index to determine
+// the proof type in the first time when this is called.
+// After that, it can be called many times without
+// rerunning the comparison.
```
The term for this is "memoized".

arlolra commented:
Is this pull meant to still be in WIP?  If so, what's left to do?  It seems pretty reasonable to me.

masomel commented:
>  Is this pull meant to still be in WIP? If so, what's left to do?

I'm doing one last pass over all the documentation to make sure we didn't miss anything. I don't think it needs to be in WIP for that though.

masomel commented:
So, I know that we had talked about not needing summaries for executable commands, but their main packages still appear in godoc, and I think it's awkward that these packages don't have a summary. What if the summary just said something like "Executable X. See README for usage information"?

arlolra commented:
Sounds reasonable.

arlolra commented:
FYI, I fixed up the go formatting issues and rebased.

c633 commented:
LGTM. Thank you @marcela @arlolra

masomel commented:
Is this ready to be merged?

c633 commented:
From my side, yes.
